### ISSUE TITLE
IE9 Placeholder Support: Larger Question around getting computed styles

### ISSUE THREAD

**Issue by zslabs** [polyfillpolyfill/polyfill-service#566]
Hi all,
I've been looking into what's stopping the placeholder polyfill from working in IE9 https://github.com/Financial-Times/polyfill-service/blob/master/polyfills/Element/prototype/placeholder/polyfill.js and all I've been able to find is the oddities around getCurrentStyle, getComputedStyle, and whether the browser supports camel-case props. Before looking into a fix for this specific polyfill, I wanted to reach out to see if there's a generic utility around this at the moment in the lib. Thanks!

[This](https://groups.google.com/d/msg/comp.lang.javascript/71ug4NBYJog/xXwUC9sgpP8J) seems to be a decent workaround, but think we could automate the camel-casing if need-be.

**Comment by triblondon** [comment]
No, I don't believe we have a utility for camelcasing (or uncamelcasing) CSS property names. I didn't write this polyfill and am not familiar with it, but if it interests you please do take a look.

**Comment by deddu** [comment]
@zslabs did you get this to work? I can't have this polyfill working with IE9 either and before I dig into the code I'd like to know your thoughts. 
To be more precise, it seems that  `Element.prototype.placeholder` remains undefined.

**Comment by deddu** [comment]
I have found a fix for the placeholder. There were two issues: 
- the `document.createElement('-ms-input')` and similar were failing due to the unsupported `-` at the beginning of the identifier. 
-  `Object.defineProperty` was missing the getter.

I'll create a PR. any feedback is welcome

**Comment by zslabs** [comment]
@deddu I hadn't had a chance yet; thanks for jumping on this!

**Comment by jonathantneal** [comment]
Great work, @deddu. I wrote the polyfill originally. Feel free to throw tomatoes or ask questions of me.

**Comment by deddu** [comment]
@jonathantneal thanks! I definitely have questions, especially because I sort of hammered at it until it worked in my VM. First I clearly would love for you to review the edits.  
Then, I did not had time to really dig into how the styles are applied, so I was just wondering if the styling logic would belong to the getter... thoughts?
I would also thank everyone for this awesome service.  I cried of happines when I found this project!! :)
